main.cpp

```diff
@@ -394,37 +456,43 @@ void Garage::repair(Truck vehcile)
 int main()
 {
     // 1.
-    Dishwasher dishwasher (1500);
-    dishwasher.washDishes(250);
-    dishwasher.lockDoor();
-    dishwasher.informUser();
-    dishwasher.doesSomethingInteresting();
-    std::cout << "**** inputVoltage : " << dishwasher.inputVoltage << std::endl;
-    dishwasher.printinputVoltage();
+    Dishwasher* dishwasher = new Dishwasher (1500);
+    DishwasherWrapper dishwasherWrapper (dishwasher);
+    dishwasher->washDishes(250);
+    dishwasher->lockDoor();
+    dishwasher->informUser();
+    dishwasher->doesSomethingInteresting();
+    std::cout << "**** inputVoltage : " << dishwasher->inputVoltage << std::endl;
+    dishwasher->printinputVoltage();
 
     // 2.
-    Bicycle bicycle;
-    bicycle.cycleAtSpeed(120);
-    bicycle.useBreak();
-    bicycle.turn("right");
-    bicycle.doesSomethingInteresting(125);
-    bicycle.doesSomethingInteresting(2);
-    std::cout << "**** Gear : " << bicycle.gear << std::endl;
-    bicycle.printGear();
+    Bicycle* bicycle = new Bicycle();
+    BicycleWrapper bicycleWrapper (bicycle);
+    bicycle->cycleAtSpeed(120);
+    bicycle->useBreak();
+    bicycle->turn("right");
+    bicycle->doesSomethingInteresting(125);
+    bicycle->doesSomethingInteresting(2);
+    std::cout << "**** Gear : " << bicycle->gear << std::endl;
+    bicycle->printGear();
+
     // 3.
-    Truck truck (120, 4565);
-    truck.moveFood(300);
-    truck.drive(140, 20);
-    truck.fitTrailer("long", 120);
-    std::cout << "Final value of X for Truck --> " << truck.doesSomethingInteresting() << std::endl;
+    Truck* truck = new Truck(120, 4565);
+    TruckWrapper truckWrapper (truck);
+    truck->moveFood(300);
+    truck->drive(140, 20);
+    truck->fitTrailer("long", 120);
+    std::cout << "Final value of X for Truck --> " << truck->doesSomethingInteresting() << std::endl;
     // 4.
-    Kitchen kitchen;
-    kitchen.start(dishwasher);
-    kitchen.stop(dishwasher);
+    Kitchen* kitchen = new Kitchen();
+    KitchenWrapper kitchenWrapper (kitchen);
+    kitchen->start(dishwasher);
+    kitchen->stop(dishwasher);
     // 5.
-    Garage garage;
-    garage.park(bicycle, truck);
-    garage.repair(truck);    
+    Garage* garage = new Garage();
+    GarageWrapper garageWrapper (garage);
+    garage->park(bicycle, truck);
+    garage->repair(truck);    
```

Comment:
This does not demonstrate using your wrapper class
review the example and revise this
Also, create your instance directly in the wrapper's constructor argument list.
Don't create it as a separate variable.
- [ ] revise